x-pack/plugins/cloud_security_posture/public/components/cloud_posture_page.tsx

```diff
@@ -267,23 +270,27 @@ export const CloudPosturePage = <TData, TError>({
       return subscriptionNotAllowedRenderer();
     }
 
-    if (getSetupStatus.isError) {
-      return defaultErrorRenderer(getSetupStatus.error);
+    if (isError) {
+      return defaultErrorRenderer(error);
     }
 
-    if (getSetupStatus.isLoading) {
+    if (isLoading) {
       return defaultLoadingRenderer();
     }
 
     /* Checks if its a completely new user which means no integration has been installed and no latest findings default index has been found */
     if (
-      getSetupStatus.data?.kspm?.status === 'not-installed' &&
-      getSetupStatus.data?.cspm?.status === 'not-installed' &&
-      getSetupStatus.data?.indicesDetails[0].status === 'empty'
+      getSetupStatus?.kspm?.status === 'not-installed' &&
+      getSetupStatus?.cspm?.status === 'not-installed' &&
+      getSetupStatus?.indicesDetails[0].status === 'empty'
     ) {
```

Comment:
This can also be added to the `getCpmStatus` function

```js
/* Checks if cspm or kspm integration has been installed and if latest findings index data has been found */
const hasCpmData = cpmStatusData?.kspm?.status !== 'not-installed' && cpmStatusData?.cspm?.status !== 'not-installed' && cpmStatusData?.indicesDetails[0].status !== 'empty'
```

And then use like this

```
/* If there's no cpmData, it's a completely new user, which means no integration has been installed and no latest findings default index has been found */
if (!hasCpmData) {
	return packageNotInstalledRenderer({ kspmIntegrationLink, cspmIntegrationLink });
}
```